src/restricted-endpoints/replication/bulk-document/bulk-document.service.ts

```diff
@@ -106,6 +107,38 @@ export class BulkDocumentService {
     };
   }
 
+  async filterFindResponse(
+    request: FindResponse,
+    user: UserInfo,
+    db: string,
+  ): Promise<FindResponse> {
+    const ability = this.permissionService.getAbilityFor(user);
+    const allDocsRequest: AllDocsRequest = {
+      keys: request.docs.map((doc) => doc._id),
+    };
+    const response = await firstValueFrom(
+      this.couchdbService.post<AllDocsResponse>(
+        db,
+        '_all_docs',
+        allDocsRequest,
+        {
+          include_docs: true,
+        },
+      ),
+    );
```

Comment:
why do we need an additional all_docs request here? Can't we just check permissions on the response of the "_find"?